sources/packages/backend/apps/api/src/esdc-integration/msfaa-integration/msfaa-response.service.ts

```diff
@@ -102,10 +103,13 @@ export class MSFAAResponseService {
   private async processReceivedRecord(
     receivedRecord: MSFAAResponseReceivedRecord,
   ): Promise<void> {
+    // The update of msfaa always comes from an external source through integration.
+    // Hence the date fields are parsed as date object from external source as their date format
+    // may not be necessarily ISO date format.
     const updateResult = await this.msfaaNumberService.updateReceivedFile(
       receivedRecord.msfaaNumber,
-      receivedRecord.borrowerSignedDate,
-      receivedRecord.serviceProviderReceivedDate,
+      getISODateOnlyString(receivedRecord.borrowerSignedDate),
```

Comment:
For these situations would not be better for the `updateReceivedFile` to receive the Date type object and then do the conversion internally?

Reply:
Changes done as we discussed.